src/ctia/schemas/utils.clj

```diff
@@ -0,0 +1,17 @@
+(ns ctia.schemas.utils
+  (:require [schema-tools.walk :as sw]
+            [schema.core :as s]))
+
+(defn recursive-open-schema-version
+  "walk a schema and replace all instances of schema version
+   with an open string schema"
+  [s]
+  (sw/walk
+   (fn [x]
+     (if (and
+          (instance? clojure.lang.IMapEntry x)
+          (= :schema_version (:k (first x))))
```

Comment:
I would replace that using `prewalk` and `map?`:

```Clojure
(defn recursive-open-schema-version
   "walk a schema and replace all instances of schema version
    with an open string schema"
   [s]
   (prewalk #(if (and (map? %)
                      (= :schema_version (:k (first x))))
              [(first x) s/Str]
              (recursive-open-schema-version %))
           m))
```

Could you add a test for the `recursive-open-schema-version` fn?

Reply:
your version gives a stackoverflow so i keep mine ;-), I added the test